tests/template/test_kedro_cli.py

```diff
@@ -205,15 +205,33 @@ def test_run_successfully_parallel_via_name(
             ParallelRunner,
         )
 
+    @pytest.mark.parametrize(
+        "config_addition,expected",
+        [
+            ({}, {}),
+            ({"params": {"foo": "baz"}}, {"foo": "baz"}),
+            ({"params": "foo:baz"}, {"foo": "baz"}),
+            ({"params": {"foo": "123.45", "baz": "678"}}, {"foo": 123.45, "baz": 678}),
+        ],
+    )
     @pytest.mark.parametrize("config_flag", ["--config", "-c"])
     def test_run_with_config(
-        self, config_flag, fake_kedro_cli, fake_load_context, fake_run_config, mocker
+        self,
+        config_addition,
+        expected,
+        config_flag,
+        fake_kedro_cli,
+        fake_load_context,
+        fake_run_config,
+        mocker,
     ):
+        config = anyconfig.load(fake_run_config)
+        config["run"].update(config_addition)
+        anyconfig.dump(config, fake_run_config)
```

Comment:
Can you explain what this is for and what are the changes you've made to this test beyond adding in more cases in the parametrisation? :)

Reply:
What I wanted to ensure with the test is that no matter which way `params` is set in the config it would be tested properly. That makes it necessary to edit the config file that is written to disk by the test fixture. However, when I do that in the fixture I can't test the case where `params` is empty (I think) because I can't propagate the actual expected outcome to the test and it also breaks other tests. Therefore, I opted to read the config from disk and extend the section I wanted then write it back to disk. A bit ugly but I am open to other options :)

Reply:
Gotcha. Had a proper look at it this morning and makes sense to me. Although I'd probably have factored that out into a fixture (that depended on the `fake_run_config` fixture). Thoughts @DmitriiDeriabinQB?

Reply:
Agree with @ZainPatelQB - making a new fixture out of it would improve the readability. I would also suggest to put the parameters logic into a separate test.

Reply:
Created a new fixture with the other one as dependency and the separated the tests. So that the original still tests the `-c` and `--config` and the new one tests the parameters. Learned something new with indirect fixtures here thanks for that :D hope that actually makes sense.